### PR TITLE
libgcrypt: bump to v1.11.

### PR DESCRIPTION
**Pull request by maleadt**

### CODE REVIEW

File: L/Libgcrypt/build_tarballs.jl [hunk 1]
```diff
@@ -33,8 +34,9 @@ products = [
 
 # Dependencies that must be installed before this package can be built
 dependencies = [
-    Dependency("Libgpg_error_jll", v"1.42.0"; compat="1.42.0"),
+    Dependency("Libgpg_error_jll", v"1.49.0"; compat="1.49.0"),
```
Comment (giordano) [inline L/Libgcrypt/build_tarballs.jl:37]:
```suggestion
    Dependency("Libgpg_error_jll"; compat="1.49.0"),
```

**Comment by maleadt** [comment]
Well, this is weird. Locally, `julia +1.7 --project build_tarballs.jl --verbose --debug i686-w64-mingw32` passes without any build error. And for FreeBSD, why doesn't it pick up the v1.50 dependency?

**Comment by maleadt** [comment]
> And for FreeBSD, why doesn't it pick up the v1.50 dependency?

Fixed by forcing v1.50, although I'm still confused why it wouldn't use this with a `compat` of v1.49...



> Locally, `julia +1.7 --project build_tarballs.jl --verbose --debug i686-w64-mingw32` passes without any build error.

I wonder if part of the host environment is leaking into the build, making it end up building different sources, or using different options? The code _is_ hidden behind a feature flag: https://github.com/gpg/libgcrypt/blob/4876a1a45c25d7e3ffb45a0af4e7c8995ce3a552/random/jitterentropy-base-user.h#L81-L98

File: L/Libgcrypt/build_tarballs.jl [hunk 2]
```diff
@@ -33,8 +40,9 @@ products = [
 
 # Dependencies that must be installed before this package can be built
 dependencies = [
-    Dependency("Libgpg_error_jll", v"1.42.0"; compat="1.42.0"),
+    Dependency("Libgpg_error_jll", v"1.50"; compat="1.49"),
```
Comment (giordano) [inline L/Libgcrypt/build_tarballs.jl:43]: Note that the lowest version of the compat is automatically used as build version, and if specifying a build version is important, having a consistent compat is usually important too
```suggestion
    Dependency("Libgpg_error_jll"; compat="1.50"),
```

Comment (maleadt) [reply L/Libgcrypt/build_tarballs.jl:43]: Thing is, the compat really is 1.49, we just only have a build for FreeBSD-aarch64 for 1.50.

**Comment by maleadt** [comment]
MWE:

```c
#include <stdint.h>
typedef uint64_t u64;

#include <time.h>

static void
jent_get_nstime(u64 *out)
{
  struct timespec tv;
  u64 tmp;

  /* On Linux we could use CLOCK_MONOTONIC(_RAW), but with
   * CLOCK_REALTIME we get some nice extra entropy once in a while
   * from the NTP actions that we want to use as well... though, we do
   * not rely on that extra little entropy.  */
  if (!clock_gettime (CLOCK_REALTIME, &tv))
    {
      tmp = tv.tv_sec;
      tmp = tmp << 32;
      tmp = tmp | tv.tv_nsec;
    }
  else
    tmp = 0;
  *out = tmp;
}
```

With Yggdrasil cloned on amdci7 and instantiating the CI Manifest, I get:

```
# cc -c test.c -o test.o
test.c: In function ‘jent_get_nstime’:
test.c:16:23: error: ‘CLOCK_REALTIME’ undeclared (first use in this function)
   if (!clock_gettime (CLOCK_REALTIME, &tv))
                       ^
test.c:16:23: note: each undeclared identifier is reported only once for each function it appears in
```

On my local system, with otherwise the exact same set-up, everything works fine...

Some part of the set-up isn't identical though, but I'm not sure how or why. Locally:

```
sandbox:${WORKSPACE}/srcdir/libgcrypt-1.11.0/random # grep CLOCK_REALTIME /opt/x86_64-linux-musl/x86_64-linux-musl/sys-root/usr/include/time.h
#define CLOCK_REALTIME           0
#define CLOCK_REALTIME_COARSE    5
#define CLOCK_REALTIME_ALARM     8
```

On Yggdrasil:

```
sandbox:${WORKSPACE}/srcdir/libgcrypt-1.11.0/random # grep CLOCK_REALTIME /opt/x86_64-linux-musl/x86_64-linux-musl/sys-root/usr/include/time.h
grep: /opt/x86_64-linux-musl/x86_64-linux-musl/sys-root/usr/include/time.h: No such file or directory
```

**Comment by maleadt** [comment]
It even reproduces with a fresh depot... Here's the verbose logs of both invocations, one on my local system, one on amdci7: https://gist.github.com/maleadt/e5764478bf01b161842ee1e0bc0edea9

**Comment by giordano** [comment]
```console
% find build/i686-w64-mingw32/yNJdRiPh/.mounts -name time.h -exec md5sum {} \; 2> /dev/null | sort                 
14a0dab31a3dea92cbbe2fd9a40c9072  build/i686-w64-mingw32/yNJdRiPh/.mounts/GCCBootstrap-x86_64-linux-musl.v4.8.5.x86_64-linux-musl.squashfs/x86_64-linux-musl/sys-root/usr/include/time.h
25b6994f6b4e75f2ce9582f5ac81af78  build/i686-w64-mingw32/yNJdRiPh/.mounts/GCCBootstrap-i686-w64-mingw32.v4.8.5.x86_64-linux-musl.squashfs/i686-w64-mingw32/sys-root/include/time.h
29e1e232fdf102bf5c56856c8a066ed8  build/i686-w64-mingw32/yNJdRiPh/.mounts/PlatformSupport-i686-w64-mingw32.v2024.8.10.x86_64-linux-musl.squashfs/i686-w64-mingw32/sys-root/include/time.h
7e6318050651a97ad6076a81e6384cb1  build/i686-w64-mingw32/yNJdRiPh/.mounts/LLVMBootstrap.v18.1.7.x86_64-linux-musl.squashfs/lib/clang/18/include/llvm_libc_wrappers/time.h
849dabb9216c8f53bbf721c4720a9356  build/i686-w64-mingw32/yNJdRiPh/.mounts/GCCBootstrap-i686-w64-mingw32.v4.8.5.x86_64-linux-musl.squashfs/i686-w64-mingw32/sys-root/include/sys/time.h
849dabb9216c8f53bbf721c4720a9356  build/i686-w64-mingw32/yNJdRiPh/.mounts/PlatformSupport-i686-w64-mingw32.v2024.8.10.x86_64-linux-musl.squashfs/i686-w64-mingw32/sys-root/include/sys/time.h
a8f8234ef9e90aae9be3252e80c1d2b6  build/i686-w64-mingw32/yNJdRiPh/.mounts/GCCBootstrap-x86_64-linux-musl.v4.8.5.x86_64-linux-musl.squashfs/x86_64-linux-musl/sys-root/usr/include/linux/time.h
a8f8234ef9e90aae9be3252e80c1d2b6  build/i686-w64-mingw32/yNJdRiPh/.mounts/PlatformSupport-x86_64-linux-musl.v2024.8.10.x86_64-linux-musl.squashfs/x86_64-linux-musl/sys-root/usr/include/linux/time.h
adeb9c7928c2fb4c364d63f7938472d9  build/i686-w64-mingw32/yNJdRiPh/.mounts/GCCBootstrap-x86_64-linux-musl.v4.8.5.x86_64-linux-musl.squashfs/x86_64-linux-musl/sys-root/usr/include/sys/time.h
```
that seems to be the problem, yes.  Interesting we never ran into this before, also not great the two artifacts step onto each other's toes.

**Comment by giordano** [comment]
Perhaps using two wildly different versions of mingw at https://github.com/JuliaPackaging/Yggdrasil/blob/ff8aa2b3b4b76a2b81e27ad73a931dec4103cb15/0_RootFS/gcc_sources.jl#L278-L279 and https://github.com/JuliaPackaging/Yggdrasil/blob/ff8aa2b3b4b76a2b81e27ad73a931dec4103cb15/0_RootFS/PlatformSupport/build_tarballs.jl#L52-L53 doesn't help.  Also not really sure why we install mingw header files twice.

**Comment by maleadt** [comment]
> Interesting we never ran into this before, also not great the two artifacts step onto each other's toes.

https://github.com/JuliaPackaging/Yggdrasil/blob/ff8aa2b3b4b76a2b81e27ad73a931dec4103cb15/M/MGARD/build_tarballs.jl#L45-L47

Also, FWIW, the `--map` arguments to `sandbox` are in the same order both times, so the nondeterminism is in `sandbox`.

**Comment by giordano** [comment]
Ah, the reference to notcurses brought me to https://github.com/JuliaPackaging/Yggdrasil/pull/4059#issuecomment-992727965 and https://github.com/JuliaPackaging/Yggdrasil/pull/4059#issuecomment-993578564.  I'm somewhat convinced that in platformsupports we should at very least match what we do in gccbootstrap, or just remove that, since I'm missing why we duplicate the work (and do it wrong).

I wonder if @staticfloat would have some opinions about this :innocent:

**Comment by staticfloat** [comment]
> I wonder if @staticfloat would have some opinions about this 😇

My opinion is that PlatformTools is very ill-defined, and instead we should be directly using the Mingw JLLs where needed, which makes it much more explicit when and where these header files are included.  BB2 does this, but progress on that front continues, despite its slowness.

**Pull request merged**